### PR TITLE
port: dialog context dialog manager prop obsolete

### LINKED ISSUE #3512: port: mark DialogManager property on DialogContext as obsolete (#5410)
The changes in [mark DialogManager property on DialogContext as obsolete (#5410)](https://github.com/microsoft/botbuilder-dotnet/pull/5410) may need to be ported to maintain parity with `microsoft/botbuilder-dotnet`.

<blockquote>
Fixes #5409
</blockquote>

Please review and, if necessary, port the changes.

### PR DESCRIPTION
**Pull request by joshgummersall**
Fixes #3512

Seems like we missed a change in `beginSkill` that fetches the conversation state from the turn context. @stevengum can you triple check that?

### CODE REVIEW

File: libraries/botbuilder-dialogs-adaptive-runtime/src/coreBot.ts [hunk 1]
```diff
@@ -48,16 +48,12 @@ export class CoreBot extends ActivityHandler {
         ResourceExtensions.useResourceExplorer(dialogManager, resourceExplorer);
         LanguageGeneratorExtensions.useLanguageGeneration(dialogManager);
         LanguageGeneratorExtensions.useLanguagePolicy(dialogManager, new LanguagePolicy(defaultLocale));
+
+        // Inserts skills dependencies into initial turn state
         SkillExtensions.useSkillClient(dialogManager, skillClient);
         SkillExtensions.useSkillConversationIdFactory(dialogManager, skillConversationIdFactory);
-        useTelemetry(dialogManager, botTelemetryClient);
 
-        /* TODO(jgummersall) reconcile:
```
Comment (joshgummersall) [inline libraries/botbuilder-dialogs-adaptive-runtime/src/coreBot.ts:55]: Verified that these are handled by the skill extensions stuff, and the `useBotState` in the `CoreBotAdapter`

File: libraries/botbuilder-dialogs-adaptive/src/actions/beginSkill.ts [hunk 2]
```diff
@@ -182,7 +182,7 @@ export class BeginSkill extends SkillDialog implements BeginSkillConfiguration {
             this.dialogOptions.connectionName = this.connectionName.getValue(dcState);
         }
         if (!this.dialogOptions.conversationState) {
-            this.dialogOptions.conversationState = dc.dialogManager.conversationState;
+            this.dialogOptions.conversationState = dc.context.turnState.get<ConversationState>('ConversationState');
```
Comment (joshgummersall) [inline libraries/botbuilder-dialogs-adaptive/src/actions/beginSkill.ts:185]: Intent is to reconcile this with https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/BeginSkill.cs#L159

Comment (stevengum) [reply libraries/botbuilder-dialogs-adaptive/src/actions/beginSkill.ts:185]: Should the JS impl throw an Exception like in .NET if a ConversationState instance isn't retrieved from the TurnState?

**Review by stevengum — APPROVED** [review]

File: libraries/botbuilder-dialogs-adaptive/src/actions/beginSkill.ts (hunk 2 above)
Comment (joshgummersall) [reply libraries/botbuilder-dialogs-adaptive/src/actions/beginSkill.ts:185]: Good question - I was following the existing implementation with respect to throwing. Should it?

Comment (stevengum) [reply libraries/botbuilder-dialogs-adaptive/src/actions/beginSkill.ts:185]: From what I can tell it definitely should; without the SkillClient, ConversationIdFactory and ConversationState the host bot won't be able to send messages, process messages or update the conversationState before forwarding activities 😅

Comment (stevengum) [reply libraries/botbuilder-dialogs-adaptive/src/actions/beginSkill.ts:185]: If beginSkill somehow works without the exceptions, then we have a diff that needs to be unpacked @chon219 have you used `BeginSkill` in Composer/AdaptiveDialogs testing?

Comment (chon219) [reply libraries/botbuilder-dialogs-adaptive/src/actions/beginSkill.ts:185]: We do have tests using `BeginSkill` but I have not tested `BeginSkill` without conversationState, in fact, `DialogManager` will throw exception in its `onTurn` if there is no conversationState found in turn state.

Comment (joshgummersall) [reply libraries/botbuilder-dialogs-adaptive/src/actions/beginSkill.ts:185]: Seems like we protect against this case further up the chain so I don't think we need another error thrown.

Comment (stevengum) [reply libraries/botbuilder-dialogs-adaptive/src/actions/beginSkill.ts]: Overall I'm okay with this, though BeginSkill is a public class and can be used outside of DialogManager. The belief that the general code-first population will not use AdaptiveDialogs and by extension BeginSkill is why I'm not opposed to this.

We should be evaluate porting these changes back to .NET or keeping a record of the difference in behavior between the implementations.

**Pull request merged**